lib/measure_repository_service_test_kit/measure_data_requirements.rb

```diff
@@ -0,0 +1,165 @@
+
+    fhir_client do
+      url :url
+    end
+
+    INVALID_ID = 'INVALID_ID'
+
+    test do
+      include DataRequirementsHelpers
+      title 'Check $data-requirements with id returns 200'
+      id 'data-requirements-01'
+      description '$data-requirements returns 200OK and Library of type module-definition when given Measure id'
+      input :measure_id, title: 'Measure id'
+
+      run do
+        fhir_operation("Measure/#{measure_id}/$data-requirements")
+
+        assert_dr_success
+      end
+    end
+
+    test do
+      include DataRequirementsHelpers
+      title 'Check $data-requirements with url returns 200'
+      id 'data-requirements-02'
+      description '$data-requirements returns 200OK and Library of type module-definition when passed in a url'
+      input :measure_url, title: 'Measure url'
+      input :measure_version, optional: true, title: 'Measure version'
+
+      run do
+        url_params_hash = {
+          resourceType: 'Parameters',
+          parameter: [{
+            name: 'url',
+            valueUrl: measure_url
+          }]
+        }
+
+        unless measure_version.nil?
+          url_params_hash[:parameter].append({ name: 'version',
+                                               valueString: measure_version })
+        end
+
+        url_params_hash = url_params_hash.freeze
+        url_params = FHIR::Parameters.new url_params_hash
+
+        fhir_operation('Measure/$data-requirements',
+                       body: url_params)
+
+        assert_dr_success
+      end
+    end
+
+    test do
+      include DataRequirementsHelpers
+      title 'Check $data-requirements with identifier returns 200'
+      id 'data-requirements-03'
+      description '$data-requirements returns 200OK and Library of type module-definition when passed in an identifier'
+      input :measure_identifier, title: 'Measure identifier'
+      run do
+        identifier_params_hash = {
+          resourceType: 'Parameters',
+          parameter: [{
+            name: 'identifier',
+            valueUrl: measure_identifier
+          }]
+        }.freeze
+
+        identifier_params = FHIR::Parameters.new identifier_params_hash
+
+        fhir_operation('Measure/$data-requirements',
+                       body: identifier_params)
+
+        assert_dr_success
+      end
+    end
+
+    test do
+      include DataRequirementsHelpers
+      title 'Check $data-requirements accepts periodStart and periodEnd parameters'
+      id 'data-requirements-04'
+      description '$data-requirements returns 200 when passed periodStart and periodEnd parameters'
+      input :measure_id, title: 'Measure id'
+
+      run do
+        fhir_operation(
+          "Measure/#{measure_id}/$data-requirements?periodStart=2019-01-01&periodEnd=2020-01-01"
+        )
+
+        assert_dr_success
+      end
+    end
+
+    test do
+      include DataRequirementsHelpers
+      title 'Check $data-requirements returns 404 for invalid measure id'
```

Comment:
```suggestion
      title 'Throws 404 error when no measure on the server matches id'
```
Nitpick but this makes it more clear that the error is thrown because the measure is not on the server, and not bc of something else like the id's format